### DISCUSSION TITLE
Home Connect integration not working

### DISCUSSION THREAD

**Discussion by sewo-x** [alexkn/node-red-contrib-homeconnect#96] [Q&A]
Hi team,

unfortunately, the integration/connection with home connect does not work for me.

I always get the following error message:
{
  "error": "unauthorized_client",
  "error_description": "client or redirection URI not authorized"
}

![image](https://github.com/alexkn/node-red-contrib-homeconnect/assets/73949083/b9bd6acc-b1d3-4ea8-83b8-1c81316c7012)

![image](https://github.com/alexkn/node-red-contrib-homeconnect/assets/73949083/92f74cd6-82b8-4360-83ff-8d658688cd9d)

Do you have any idea to solve this?

**Comment by alexkn** [comment]
The Redirect URI in the developer portal must be identical to the callback URL in the Auth Node.

**Comment by sewo-x** [comment]
Thanks....sometimes it's that easy.

Now I was one step closer. But now I have the following message:
![image](https://github.com/alexkn/node-red-contrib-homeconnect/assets/73949083/16338c57-b8eb-4444-9fd5-b904ddb72528)

![image](https://github.com/alexkn/node-red-contrib-homeconnect/assets/73949083/25c324c4-ac5b-4ddf-b7e1-cac6b6477ea9)

**Comment by alexkn** [comment]
I'm not sure. Try using the local URL instead of my.homeassistant.io

**Reply by sewo-x** [reply]
Thanks for your feedback.
I tried this several times, but receiving the following error while using a local URL in home connect developer portal:
![image](https://github.com/alexkn/node-red-contrib-homeconnect/assets/73949083/5c452dc7-97a2-46c7-901e-9b87935c6739)

**Reply by sewo-x** [reply]
I am using a duckdns based on my IP:
 
Receiving the following URL with site not found:
`http://XXXXXXXX.duckdns.org:1880/homeconnect/auth/callback?code=eyJyZWdpb24iOiJFVSIsInN0YWdlIjoiUFJEIiwiY3JlZiI6IkJFMzI2N0Y0IiwidG9rZW4iOiI4M2YyNzExMy0yNjEwLTRjYzUtOGYyYy1hNGVjNGQ4ZTQzMDIiLCJjbHR5IjoicHJpdmF0ZSIsIngtcmVnIjoiRVUiLCJ4LWVudiI6IlBSRCJ9&grant_type=authorization_code`

**Comment by sewo-x** [comment]
Any idea?
